src/script/smoothenXYZData.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * (c) 2023, Center for Computational Thinking and Design at Aarhus University and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+interface XYZData {
+  x: number[];
+  y: number[];
+  z: number[];
+}
+
+const smoothen = (d: number[]): number[] => {
+  if (d.length === 0) {
+    return d;
+  }
+  const newData: number[] = [];
+  let prevValue = d[0];
```

Comment:
This is a little different to the InputBehaviour.ts smoothening data logic in that the initial prevValue here is the first value instead of 0 ([see here for InputBehaviour.ts code](https://github.com/microbit-foundation/ml-trainer/blob/02c2599592699b72560429a08f92592a740a4abb/src/script/connection-behaviours/InputBehaviour.ts#L76-L78)). This is to prevent the slope start from 0 in the beginning of the recording graphs (see image for reference):
<img width="669" alt="Screenshot 2024-01-31 at 10 56 18" src="https://github.com/microbit-foundation/ml-trainer/assets/145345672/a443d56d-eb0d-453e-b604-7452fcbf0900">